web/modules/custom/acceptance/src/Form/DeliverDigestMessagesForm.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\joinup_acceptance\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Queue\QueueFactory;
+use Drupal\Core\Queue\QueueWorkerManagerInterface;
+use Drupal\Core\Queue\RequeueException;
+use Drupal\Core\Queue\SuspendQueueException;
+use Drupal\message_digest\Traits\MessageDigestTrait;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Form allowing to deliver digest messages.
+ */
+class DeliverDigestMessagesForm extends FormBase {
+
+  use MessageDigestTrait;
+
+  /**
+   * The queue service.
+   *
+   * @var \Drupal\Core\Queue\QueueFactory
+   */
+  protected $queueFactory;
+
+  /**
+   * The queue manager.
+   *
+   * @var \Drupal\Core\Queue\QueueWorkerManagerInterface
+   */
+  protected $queueManager;
+
+  /**
+   * Constructs a new form instance.
+   *
+   * @param \Drupal\Core\Queue\QueueFactory $queue_factory
+   *   The queue service.
+   * @param \Drupal\Core\Queue\QueueWorkerManagerInterface $queue_manager
+   *   The queue manager.
+   */
+  public function __construct(QueueFactory $queue_factory, QueueWorkerManagerInterface $queue_manager) {
+    $this->queueFactory = $queue_factory;
+    $this->queueManager = $queue_manager;
+
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): self {
+    return new static(
+      $container->get('queue'),
+      $container->get('plugin.manager.queue_worker')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+    $form['deliver'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Deliver message digests'),
+      '#submit' => [[$this, 'deliverDigestMessages']],
+    ];
+    return $form;
+  }
+
+  /**
+   * Provides a submit handler for digest messages delivery button.
+   *
+   * @param array $form
+   *   The form API render array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state API object.
+   */
+  public function deliverDigestMessages(array &$form, FormStateInterface $form_state): void {
+    $this->expireDigestMessages();
+    do {
+      $this->expireMessageDigestNotifiers();
+      message_digest_cron();
+      $this->processQueue();
+    } while ($this->countAllUndeliveredDigestMessages());
+  }
```

Comment:
There is no indication that a process is completed. We should add a status message. Since the `::countAllUndeliveredDigestMessages()` returns the count of messages to be delivered, we could even show the number of digest messages sent. But even a "Process complete" success message should be fine.

Reply:
OK, then, at least we should also say something on queue error.